Source/JavaScriptCore/API/JSClassRef.cpp

```diff
@@ -116,40 +177,51 @@ Ref<OpaqueJSClass> OpaqueJSClass::create(const JSClassDefinition* clientDefiniti
 
     JSClassDefinition protoDefinition = kJSClassDefinitionEmpty;
     protoDefinition.finalize = 0;
+    protoDefinition.version = clientDefinition->version;
     std::swap(definition.staticFunctions, protoDefinition.staticFunctions); // Move static functions to the prototype.
-    
+
     // We are supposed to use JSClassRetain/Release but since we know that we currently have
     // the only reference to this class object we cheat and use a RefPtr instead.
     RefPtr<OpaqueJSClass> protoClass = adoptRef(new OpaqueJSClass(&protoDefinition, 0));
-    return adoptRef(*new OpaqueJSClass(&definition, protoClass.get()));
+    OpaqueJSClass* clazz = new OpaqueJSClass(&definition, protoClass.get());
+    protoClass->prototypeForClass = JSClassRetain(clazz);
+    return adoptRef(*clazz);
```

Comment:
Append comment to above comment if this is correct usage, keep upstream comment.